payroll-processor-client/src/environments/environment.local.sample.ts

```diff
@@ -0,0 +1,6 @@
+import { Environment } from './environment-types';
+
+export const environment: Environment = {
```

Comment:
We could "merge" defaults with local settings if we wanted, by rewriting this file as:

```
import { Environment } from './environment-types';
import { environment as envSource } from './environment';

export const environment: Environment = {
  ...envSource,
  production: false,
  domain: 'http://localhost:7071',
};
```

That way we could put the defaults in `environment.ts` and anything not explicitly overwritten in `environment.local.ts` would flow through from there.

I've never done this on a project before, but I have wondered if it would be beneficial...

If we did that, there might be better names for the files, like:

`environment.local.user.sample.ts` - user specific local sample (committed)
`environment.local.user.ts` - user specific local configuration copied from `environment.local.user.sample.ts) (gitignored)
`environmemt.local.ts` - default local (committed)

Thoughts?

Reply:
I like going the suggested route, this provides a working solution out of the box for the repository but allows users to customize their environment fairly easily.